video/out/w32_common.c

```diff
@@ -1560,20 +1579,30 @@ static char **get_disp_names(struct vo_w32_state *w32)
 static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
 {
     switch (request) {
-    case VOCTRL_FULLSCREEN:
-        if (w32->opts->fullscreen != w32->current_fs)
-            reinit_window_state(w32);
+    case VOCTRL_VO_OPTS_CHANGED: {
+        void *changed_option;
+
+        while (m_config_cache_get_next_changed(w32->opts_cache,
+                                               &changed_option))
+        {
+            struct mp_vo_opts *vo_opts = w32->opts_cache->opts;
+
+            if (changed_option == &vo_opts->fullscreen &&
+                vo_opts->fullscreen != w32->current_fs)
```

Comment:
Not sure if the second check is really needed. vo_opts->fullscreen can only change due to calling m_config_cache_get_next_changed. And also, the function checks whether the value actually changed (compared to vo_opts->fullscreen). So if you do "vo_opts->fullscreen = w32->current_fs;" whenever current_fs changes, then this second check is redundant. (If that's correct, having the check is not actively harmful but might be confusing.)

Reply:
Removed the check → seemed to still work (since the update is in that spot atm).
Moved the variable update to the one place where `w32->current_fs` is set to a value → still works.

I think this is now applied. Additionally, I would almost say that one could just remove `w32->current_fs` at this point?